File: 2022/r/day01.r
# --- Day 1: Calorie Counting ---

input <- as.numeric(readLines("2022/data/input.txt"))

# Part 1
sort(sapply(split(input, cumsum(is.na(input))), sum, na.rm=T), TRUE)[1]

# Part 2
sum(sort(sapply(split(input, cumsum(is.na(input))), sum, na.rm=T), TRUE)[1:3])

File: 2022/julia/day01.jl
# --- Day 1: Calorie Counting ---
input = readlines("2022/data/input01.txt")

function per_elf_sums(input)
    sums = [0]
    for x in input
        x == "" ? push!(sums, 0) : sums[end] += parse(Int, x)
    end
    sums
end

sums = per_elf_sums(input)

println("Answer 1: ", maximum(sums))
println("Answer 2: ", sum(sort(sums, rev=true)[1:3]))
